cmd/argocd/commands/app.go

```diff
@@ -1049,24 +1051,55 @@ func NewApplicationDeleteCommand(clientOpts *argocdclient.ClientOptions) *cobra.
 				c.HelpFunc()(c, args)
 				os.Exit(1)
 			}
+
 			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
 			defer util.Close(conn)
+
 			for _, appName := range args {
 				appDeleteReq := applicationpkg.ApplicationDeleteRequest{
 					Name: &appName,
 				}
 				if c.Flag("cascade").Changed {
 					appDeleteReq.Cascade = &cascade
 				}
+
 				_, err := appIf.Delete(context.Background(), &appDeleteReq)
 				errors.CheckError(err)
+
+				if c.Flag("wait").Changed {
+					err = waitForAppDeletion(appIf, appName, timeout)
+					errors.CheckError(err)
+				}
 			}
 		},
 	}
 	command.Flags().BoolVar(&cascade, "cascade", true, "Perform a cascaded deletion of all application resources")
+	command.Flags().BoolVar(&wait, "wait", false, "Will cause the command to only return once the application has been deleted")
+	command.Flags().IntVar(&timeout, "timeout", 60, "Time to spend waiting for application to delete, default 60s")
```

Comment:
Can I please check this is consistent with `app create`?

Reply:
Will do

Reply:
Also keep an eye out on this:

https://github.com/argoproj/argo-cd/blob/73590e1a3982501865577c2a2c9ec8794391b813/cmd/argocd/commands/app.go#L1852